packages/core/src/hooks/data/useUpdate.ts

```diff
@@ -149,60 +146,97 @@ export const useUpdate = <
         },
         {
             onMutate: async ({ resource, id, mutationMode, values }) => {
-                const previousQueries: ContextQuery[] = [];
-
-                const allQueries = getAllQueries(resource, id);
+                const previousQueries: PreviousQuery<TData>[] =
+                    queryClient.getQueriesData([resource]);
 
                 const mutationModePropOrContext =
                     mutationMode ?? mutationModeContext;
 
-                for (const queryItem of allQueries) {
-                    const { queryKey } = queryItem;
-                    await queryClient.cancelQueries(queryKey, undefined, {
-                        silent: true,
-                    });
+                await queryClient.cancelQueries(resource, undefined, {
+                    silent: true,
+                });
 
-                    const previousQuery =
-                        queryClient.getQueryData<QueryResponse<TData>>(
-                            queryKey,
-                        );
+                if (!(mutationModePropOrContext === "pessimistic")) {
+                    if (previousQueries) {
+                        const listQueries = queryClient.getQueriesData([
+                            resource,
+                            "list",
+                        ]);
+
+                        if (listQueries.length > 0) {
+                            queryClient.setQueriesData(
+                                [resource, "list"],
+                                (previous?: GetListResponse<TData> | null) => {
+                                    if (!previous) {
+                                        return null;
+                                    }
+                                    const data = previous.data.map(
+                                        (record: TData) => {
+                                            if (
+                                                record.id?.toString() ===
+                                                id.toString()
+                                            ) {
+                                                return {
+                                                    id: Number(id),
```

Comment:
```suggestion
                                                    id: Number(id),
```
why do we convert id to number? id can be a string